cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -313,37 +313,49 @@ impl LockedManifestCatalog {
         seed_lockfile: Option<&LockedManifestCatalog>,
         client: &impl catalog::ClientTrait,
     ) -> Result<LockedManifestCatalog, LockedManifestError> {
-        let groups = Self::collect_package_groups(manifest, seed_lockfile).collect();
-
-        // lock existing packages
+        let groups = Self::collect_package_groups(manifest, seed_lockfile);
+        let (already_locked_packages, groups_to_lock) =
+            Self::split_fully_locked_groups(groups, seed_lockfile);
+
+        if groups_to_lock.is_empty() {
+            debug!("All packages are already locked, skipping resolution");
```

Comment:
Suggestion: for all new logging let's default to `tracing` macros so that there's less to switch over whenever we migrate from `log::*` to `tracing::*`

Reply:
i think its easier to keep using `debug!` as is and just change the imports
i.e. globally replace `log::` with `tracing::`.
But i admit we haven't been very consistent with either.